File: apps/api/app/api_client/endpoints/templates.py
"""API client for template endpoints with typed responses."""

from app.api_client.client import api_client
from api.schemas.template import TemplateDetail, TemplateListResponse


class TemplatesAPI:
    """API client for template endpoints with typed responses."""

    @staticmethod
    async def list_resume_templates() -> TemplateListResponse:
        """List all available resume templates. Returns TemplateListResponse model."""
        return await api_client.get(
            "/api/v1/templates/resumes",
            response_model=TemplateListResponse,
        )

    @staticmethod
    async def get_resume_template(template_id: str) -> TemplateDetail:
        """Get a specific resume template. Returns TemplateDetail model."""
        return await api_client.get(
            f"/api/v1/templates/resumes/{template_id}",
            response_model=TemplateDetail,
        )

    @staticmethod
    async def list_cover_letter_templates() -> TemplateListResponse:
        """List all available cover letter templates. Returns TemplateListResponse model."""
        return await api_client.get(
            "/api/v1/templates/cover-letters",
            response_model=TemplateListResponse,
        )

    @staticmethod
    async def get_cover_letter_template(template_id: str) -> TemplateDetail:
        """Get a specific cover letter template. Returns TemplateDetail model."""
        return await api_client.get(
            f"/api/v1/templates/cover-letters/{template_id}",
            response_model=TemplateDetail,
        )

File: apps/web/src/lib/api/templates.ts
/**
 * Templates API client endpoints.
 */

import { apiRequest } from "./client";
import type { Template } from "@resume/database/types";

export const templatesAPI = {
  /**
   * List all available resume templates.
   */
  listResumeTemplates: (): Promise<Template[]> => {
    return apiRequest<Template[]>("/api/templates?type=resume");
  },

  /**
   * List all available cover letter templates.
   */
  listCoverLetterTemplates: (): Promise<Template[]> => {
    return apiRequest<Template[]>("/api/templates?type=cover_letter");
  },

  /**
   * List all templates.
   */
  list: (): Promise<Template[]> => {
    return apiRequest<Template[]>("/api/templates");
  },

  /**
   * Get a specific template by ID.
   */
  get: (templateId: number): Promise<Template> => {
    return apiRequest<Template>(`/api/templates/${templateId}`);
  },
};
